plugins/decoder.rst

```diff
@@ -1,4 +1,34 @@
 Decoder Plugin
 ==============
 
-TODO
+Decoder Plugin decodes the body of the response with filters coming from the `Transfer-Encoding` or `Content-Encoding`
+headers::
+
+    use Http\Discovery\HttpClientDiscovery;
+    use Http\Client\Plugin\PluginClient;
+    use Http\Client\Plugin\DecoderPlugin;
+
+    $decoderPlugin = new DecoderPlugin();
+
+    $pluginClient = new PluginClient(
+        HttpClientDiscovery::find(),
+        [$decoderPlugin]
+    );
+
+Actually it can decodes 4 different filters:
+
+ * chunked: Decode a stream with a chunked encoding (no size in the `Content-Length` header of the response)
+ * compress: Decode a stream encoded with the compress filter according to `RFC 1950`_
+ * Deflate: Decode a stream encoded with the inflate filter according to `RFC 1951`_
+ * gzip: Decode a stream encoded with the gzip filter according to `RFC 1952`_
+
+You can also use the decoder plugin to only decodes the `Transfer-Encoding` header and not the `Content-Encoding` one
+by setting the first parameter of the constructor to false::
+
+    $decoderPlugin = new DecoderPlugin(false);
+
+This is useful when you want to get the encoded response body, or acting as a proxy.
```

Comment:
is "This" referring to only handle Transfer-Encoding but not Content-Encoding? or to the whole DecoderPlugin?